flax/nn/base.py

```diff
@@ -943,7 +945,13 @@ def store(self, value):
     frame = _top_frame('store')
     if not self._mutable:
       raise ValueError('Collection is not mutable. Use the `mutate` method to'
-                       'create a mutable copy.')
+                       ' create a mutable copy.')
+    master = utils._tracer_of_value(value)
```

Comment:
I'd like to see a comment before this block. I can't follow what Tracers do when reading the JAX code that lacks comments.

Is there a 2-3 sentence description of what JAX tracers hold and how this test works?

Or are they documented anywhere that I'm missing? (I also searched in readthedocs)